fsspec/dircache.py

```diff
@@ -1,9 +1,17 @@
+from __future__ import annotations
 import time
-from collections.abc import MutableMapping
+from typing import MutableMapping, List, TypedDict, Iterator, Any
+
 from functools import lru_cache
 
 
-class DirCache(MutableMapping):
+class DirEntry(TypedDict):
+    name: str
+    size: int
+    type: str
```

Comment:
See https://peps.python.org/pep-0589/ for more details.

Reply:
More fields are typically present, these are the required ones.

Reply:
Hm.. I can just make it a `Dict[str, Any]` for now then? I think technically `TypedDict` supports having extra keys, but I've run into a bunch of weird limitations if you actually try to use it like that.

Reply:
Yes please, simple is good